src/kakarot/execution_context.cairo

```diff
@@ -487,18 +487,22 @@ namespace ExecutionContext {
 
     // @notice Update the array of contracts to destroy.
     // @param self The pointer to the execution context.
-    // @param destroy_contracts_len Array new length.
-    // @param destroy_contracts The pointer to the new array of contracts.
-    func update_destroy_contracts(
+    // @param destroy_contracts_len Array length of destroy_contracts to add.
+    // @param destroy_contracts The pointer to the new array of contracts to destroy.
+    func add_n_destroy_contracts(
         self: model.ExecutionContext*,
         destroy_contracts_len: felt,
         destroy_contracts: felt*,
-        stop: felt,
     ) -> model.ExecutionContext* {
+        Helpers.fill_array(
+            fill_len=destroy_contracts_len,
+            input_arr=destroy_contracts,
+            output_arr=self.destroy_contracts + self.destroy_contracts_len,
+        );
         return new model.ExecutionContext(
             call_context=self.call_context,
             program_counter=self.program_counter,
-            stopped=stop,
+            stopped=self.stopped,
```

Comment:
`stop=TRUE` as per your previous comments?

Reply:
Since this one is called to append sub_ctx.destroy_contracts to the calling context I think it shouldn't modify stopped. I am only setting it to true with the single item push_to_destroy_contract function called in exec_selfdestruct

Reply:
but as soon as there's something in this array it should be stopped right? Consequently I would enforce the stopped=TRUE otherwise there's a risk that it's actually never stopped (if only using `add_n`, even with n=1)

Also, it's bug prone to have two functions add_1/add_n not behaving the same (imagine someone makes a refacto one day to use it in exec_selfdestruct)

then go to go!

Reply:
I understood that current context should be stopped but not the calling one and it's the calling context that is being updated here. We only want to save contract to destroy from sub_ctx so they can be destroyed at the end of the whole tx. Maybe I can put `stopped` as a parameter to both function and add comment on the wanted behavior so it gets less confusing

Reply:
ok let's keep it like that for now